staging/src/k8s.io/kubectl/pkg/cmd/get/table_printer.go

```diff
@@ -45,24 +46,26 @@ func (t *TablePrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
 	return t.Delegate.PrintObj(obj, writer)
 }
 
+var recognizedTableVersions = map[schema.GroupVersionKind]bool{
+	metav1beta1.SchemeGroupVersion.WithKind("Table"): true,
+	metav1.SchemeGroupVersion.WithKind("Table"):      true,
+}
+
 func decodeIntoTable(obj runtime.Object) (runtime.Object, error) {
 	event, isEvent := obj.(*metav1.WatchEvent)
 	if isEvent {
 		obj = event.Object.Object
 	}
 
-	if obj.GetObjectKind().GroupVersionKind().Group != metav1beta1.GroupName {
-		return nil, fmt.Errorf("attempt to decode non-Table object into a v1beta1.Table")
-	}
-	if obj.GetObjectKind().GroupVersionKind().Kind != "Table" {
-		return nil, fmt.Errorf("attempt to decode non-Table object into a v1beta1.Table")
+	if !recognizedTableVersions[obj.GetObjectKind().GroupVersionKind()] {
+		return nil, fmt.Errorf("attempt to decode non-Table object")
 	}
 
 	unstr, ok := obj.(*unstructured.Unstructured)
 	if !ok {
 		return nil, fmt.Errorf("attempt to decode non-Unstructured object")
 	}
-	table := &metav1beta1.Table{}
+	table := &metav1.Table{}
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstr.Object, table); err != nil {
```

Comment:
Huh.  I had thought we actually got a real object back.  Technically this is more expensive than it has to be because of that.

This also makes the assumption metav1 to metav1beta1 are equiv, which I don't really think is safe.  I'd expect something in this code block to statically assert that (maybe `var _ metav1.Table = metav1beta1.Table`) just on the off chance it breaks and we forget.

Reply:
(safe in the sense of "will we remember")

Reply:
added type-casts as tests